system/modules/core/elements/ContentDownload.php

```diff
@@ -88,7 +88,14 @@ protected function compile()
 		/** @var \PageModel $objPage */
 		global $objPage;
 
-		$strHref = $objPage->getFrontendUrl() . (Environment::get('queryString') ? '?' . Environment::get('queryString') : '');
```

Comment:
Makes no difference at all IMO. Except now the link actually works in the BE and it does not in your suggested code.